ts/client/scripts/rebalancer.ts

```diff
@@ -0,0 +1,175 @@
+}
+
+async function rebalancer(): Promise<void> {
+  // Load client
+  const options = AnchorProvider.defaultOptions();
+  const connection = new Connection(CLUSTER_URL!, options);
+  const user = Keypair.fromSecretKey(
+    Buffer.from(
+      JSON.parse(
+        process.env.KEYPAIR || fs.readFileSync(USER_KEYPAIR!, 'utf-8'),
+      ),
+    ),
+  );
+  const userWallet = new Wallet(user);
+  const userProvider = new AnchorProvider(connection, userWallet, options);
+  const client = await MangoClient.connect(
+    userProvider,
+    CLUSTER,
+    MANGO_V4_ID[CLUSTER],
+    {
+      idsSource: 'get-program-accounts',
+    },
+  );
+
+  // Load mango account
+  let mangoAccount = await client.getMangoAccount(
+    new PublicKey(MANGO_ACCOUNT_PK),
+    true,
+  );
+  console.log(
+    `MangoAccount ${mangoAccount.publicKey} for user ${user.publicKey} ${
+      mangoAccount.isDelegate(client) ? 'via delegate ' + user.publicKey : ''
+    }`,
+  );
+  await mangoAccount.reload(client);
+
+  // Load group
+  const group = await client.getGroup(mangoAccount.group);
+  await group.reloadAll(client);
+  const usdcBank = group.getFirstBankByMint(
+    new PublicKey('EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'),
+  );
+
+  // Loop indefinitely
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    await group.reloadAll(client);
+    mangoAccount = await mangoAccount.reload(client);
+    // console.log(mangoAccount.toString(group, true));
+
+    for (const tp of mangoAccount
+      .tokensActive()
+      .filter((tp) => tp.tokenIndex !== usdcBank.tokenIndex)) {
+      const baseBank = group.getFirstBankByTokenIndex(tp.tokenIndex);
+      const tokenBalance = tp.balanceUi(baseBank);
+
+      const serum3Markets = Array.from(
+        group.serum3MarketsMapByMarketIndex.values(),
+      )
+        // Find correct $TOKEN/$USDC market
+        .filter(
+          (serum3Market) =>
+            serum3Market.baseTokenIndex === tp.tokenIndex &&
+            serum3Market.quoteTokenIndex === usdcBank.tokenIndex,
+        );
+      if (!serum3Markets) {
+        continue;
+      }
+      const serum3Market = serum3Markets[0];
+      const serum3MarketExternal = group.serum3ExternalMarketsMap.get(
+        serum3Market.serumMarketExternal.toBase58(),
+      )!;
+      const maxBaseQuantity = serum3MarketExternal.baseSizeNumberToLots(
+        Math.abs(tokenBalance),
+      );
+      // Skip if quantity is too small
+      if (maxBaseQuantity.eq(new BN(0))) {
+        console.log(
+          ` - Not rebalancing ${tokenBalance} $${baseBank.name}, quantity too small`,
+        );
+        continue;
+      }
+      console.log(`- Rebalancing ${tokenBalance} $${baseBank.name}`);
+
+      const price = await serum3Market.computePriceForMarketOrderOfSize(
+        client,
+        group,
+        Math.abs(tokenBalance),
+        tokenBalance > 0 ? 'sell' : 'buy',
+      );
+      try {
+        const sig = await client.serum3PlaceOrder(
```

Comment:
closing oo in same tx, gives this https://explorer.solana.com/tx/2y9tA426pRC28yxqoZSFFfrhSvRoo1UhUPRSZkr7R5ZZJMVeruJfKkSHQN2Q9kpUNLaxYc3Tbm5f5MZG4NUV4F6U, not sure what is wrong, now I have a separate bulk closure of all oo accounts below separately which I confirmed to work